.gitattributes

```diff
@@ -53,6 +53,8 @@ Documentation/ReleaseNotes/* hooks-max-size=300000
 Utilities/Doxygen/doxygen.config.in hooks-max-size=150000
 
 Modules/ThirdParty/GDCM/src/gdcm/Source/DataDictionary/gdcmPrivateDefaultDicts.cxx hook-max-size=1021501
+Modules/ThirdParty/GDCM/src/gdcm/Source/DataDictionary/gdcmDefaultDicts.cxx hook-max-size=1200000
```

Comment:
Should this be `hook-max-size` or `hooks-max-size`?

Reply:
That is what I am testing, `hook-max-size` was used already in .gitattributes

Reply:
Yeah, it's the plural, I am fixing all of them `hook`->`hooks`

However, still one ghostflow error that I think must be fixed upstream.

Reply:
Option 1: merge with the known upstream problem, and fix in upstream later.
Option 2: fix in upstream first, then re-run the update here in ITK.
Option 3: ignore the upstream problem 😞

Reply:
I don't think is upstream, upstream doesn't know about hooks-max-size.

I am trying to modify the "local" to gdcm .gitattributes for that path, but doesn't seem to work.

Reply:
The one you are modifying only applies after merging. The GDCM subtree before the merge only cares about subtree's `.gitattributes`, which come from [the upstream](https://github.com/malaterre/GDCM/blob/master/.gitattributes).

Reply:
I see, however, ITK does modify that file on top of the upstream .gitattributes. See https://github.com/InsightSoftwareConsortium/ITK/blob/42719cad6626ca623b62b94f1e961ffeb69e4d2e/Modules/ThirdParty/GDCM/src/gdcm/.gitattributes

Reply:
Maybe the paths there should not start with a `/`?

Reply:
I tried with no effect.

Reply:
I am out of ideas then. Maybe @bradking can provide some insight?

Reply:
The ghostflow check is complaining about the *import* commit, not the after-merged file.  The import script itself needs to add a `.gitattributes` entry for the path.

Reply:
In other words, the import script needs to pretend that the upstream content had the `.gitattributes` entry, but actually add it itself before creating the import commit.

Reply:
Thanks @bradking, so if I understood correctly, we can maybe hack it in the UpdateFromUpstream.sh script:

```bash
extract_source () {
    git_archive
    pushd "${extractdir}/${name}-reduced"
    echo "* -whitespace" > .gitattributes
    echo "Source/DataDictionary/gdcmDefaultDicts.cxx hooks-max-size=1200000" > .gitattributes
    popd
}

```

Reply:
> PR upstream adding the path to .gitattributes

I think that is the best solution, which we can do later if it will take a long time for that round-trip.

Reply:
Setting the attribute in upstream will look strange because it has no need for this.

Several of our import scripts in ITK, VTK, CMake, etc. all have little attribute modification steps like this.  The `>>.gitattributes` line looks fine.

Reply:
> The >>.gitattributes line looks fine.

I am missing then how to avoid the merge conflict. It seems a trivial one.. but maybe there is a merge strategy set up?

Reply:
Don't add the attribute on the ITK side ahead of time.  Or, if it is there already, manually resolve the conflict once.  Future updates should merge cleanly.

Reply:
Fixing the trivial merge conflict manually worked. Thanks Brad.